Feature/workspace (#68)
geobeyond/geoserverx: docs/pages/sync/workspace.md

# Workspaces 

`geoserverx` allows users to access all/one workspace from GeoServer, along with ability to do CRUD operations on workspaces. 

!!! get "Get started"

>     To start using `geoserverx` in Sync mode, create a new instance of `SyncGeoServerX` Class, ream more about it [here](https://geobeyond.github.io/geoserverx/pages/sync/)

Reviewer: ```suggestion
    To start using `geoserverx` in Sync mode, create a new instance of `SyncGeoServerX` Class, read more about it [here](https://geobeyond.github.io/geoserverx/pages/sync/)
```

```

## Get single workspace
This command fetches workspace with parameter as name of it from GeoServer.
```Python
# Get workspace with name `cite`
client.get_workspace('cite')
```

## Create workspace
This command allows user to create new workspace. 

| Parameter | Required | Default value | Data type      | Description                          |
| ----------- | ----------- | ----------- |---- | ------------------------------------ |
| Name | :white_check_mark: | | `str`      |  To define Name of the workspace  |
| default | :negative_squared_cross_mark: | `False` | `bool`      |  To define whether to keep workspace as default or not |

> | Isolated | :negative_squared_cross_mark: | `False` |`bool`      |  To define whether to keep workspace as default or not |

Reviewer: Same comment as above, description for `Isolated` is not correct. Also, as an additional nitpick, please use the actual name of each parameter, including the correct casing (_i.e._ don't capitalize the words)

> | Name | :white_check_mark: | | `str`      |  To define Name of the workspace  |
> | default | :negative_squared_cross_mark: | `False` | `bool`      |  To define whether to keep workspace as default or not |
> | isolated | :negative_squared_cross_mark: | `False` |`bool`      |  To define whether to keep workspace as default or not |

Reviewer: ```suggestion
| Name | :white_check_mark: | | `str`      |  workspace name  |
| default | :negative_squared_cross_mark: | `False` | `bool`      |  whether to keep workspace as default |
| isolated | :negative_squared_cross_mark: | `False` |`bool`      |  set workspace as being isolated |
```

  
```Python
#Create new workspace with name `my_wrkspc` , make it Default and Isolated
client.create_workspace(name='my_wrkspc',default=True,Isolated=True)
```

## Delete workspace
This command allows user to delete  workspace. 

| Parameter | Required |   Default value | Data type      | Description                          |
| ----------- | ----------- | ----------- | -- | ------------------------------------ |
| workspace | :white_check_mark: | |  `str`      |  Name of the workspace  |
| recurse | :negative_squared_cross_mark: | `False` |  `bool`      |  This parameter recursively deletes all layers referenced by the specified workspace, including data stores, coverage stores, feature types, and so on |

```Python
#Delete workspace with name `my_wrkspc`.
client.delete_workspace(workspace='my_wrkspc',recurse=True)
```

## Update workspace
This command allows user to update existing workspace. 

| Parameter | Required |  Data type      | Description                          |
| ----------- | ----------- | ----------- |  ------------------------------------ |
| name | :white_check_mark: |   `str`      |  Name of the workspace to be updated  |

> | update | :negative_squared_cross_mark:  |  [`UpdateWorkspaceInfo`](https://github.com/geobeyond/geoserverx/blob/b7757c9f0130864b06c40c2faa17afc841fc705f/src/geoserverx/models/workspace.py#L42)      | To define body of the update request |

Reviewer: The `update` parameter is required
